src/autoscaler/metricscollector/config/config.go

```diff
@@ -126,6 +132,9 @@ func (c *Config) Validate() error {
 		return fmt.Errorf("Configuration error: RefreshInterval is 0")
 	}
 
+	if (c.Collector.CollectMethod != CollectMethodPolling) && (c.Collector.CollectMethod != CollectMethodStreaming) {
```

Comment:
We may add firehose in the future...
but I guess we are not going to provide multiple ways .. So far my thinking is they are just for internal use only since we are not sure which is the best way given we have not done performance test yet... 

Polling will eventually be dropped since it can only provide memory ( and cpu ) support.